zeebe/snapshot/src/test/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStoreTest.java

```diff
@@ -242,6 +245,21 @@ public void shouldNotDeleteCorruptSnapshotWithoutValidSnapshot() throws IOExcept
     assertThat(corruptSnapshot.getChecksumPath()).exists();
   }
 
+  @Test
+  public void shouldUseChecksumProviderForChecksumsIfSupplied() throws IOException {
+    final Map<String, Long> badChecksums = new HashMap<>();
+    badChecksums.put(SNAPSHOT_CONTENT_FILE_NAME, 123L);
+    final var testChecksumProvider = new TestChecksumProvider(badChecksums);
+
+    final var store = new FileBasedSnapshotStore(1, rootDirectory, testChecksumProvider);
+    scheduler.submitActor(store).join();
+
+    final var takenSnapshot = (FileBasedSnapshot) takeTransientSnapshot(1, store).persist().join();
+
+    final var persistedChecksums = SnapshotChecksum.read(takenSnapshot.getChecksumPath());
+    assertThat(persistedChecksums.getChecksums().get(SNAPSHOT_CONTENT_FILE_NAME)).isEqualTo(123L);
+  }
+
```

Comment:
:upside_down_face: please split this up into given, when and then. You can take a look at the other tests for examples.